hardhat.config.ts

```diff
@@ -26,11 +26,17 @@ const homestead = {
 
 const config: HardhatUserConfig = {
   solidity: {
-    version: "0.8.15",
+    version: "0.8.19",
     settings: {
       optimizer: {
         enabled: true,
-        runs: 200,
+        runs: 9999999,
+      },
+      metadata: {
+        // do not include the metadata hash, since this is machine dependent
+        // and we want all generated code to be deterministic
+        // https://docs.soliditylang.org/en/v0.7.6/metadata.html
+        bytecodeHash: 'none',
```

Comment:
I'm not sure I understand why we would want this change.  From the docs it would seeem we want to include the btyecodeHash "to more safely interact with the contract and verify its source code." https://docs.soliditylang.org/en/v0.8.19/metadata.html
Is there an issue with including it?

Reply:
An example in the wild from the Uniswap codebase. https://github.com/Uniswap/v3-core/blob/d8b1c635c275d2a9450bd6a78f3fa2484fef73eb/hardhat.config.ts#L60
Solidity bytecodeHash depends on the contract source code hash and not the AST so even a difference of a whitespace changes the entire bytecode (considering tableland is on mulitple chains). Solc [`0.8.18`](https://github.com/ethereum/solidity/releases/tag/v0.8.18) added `--no-cbor-metadata` natively to disable this.

Reply:
ah, makes sense. i would be ideal to have deterministic deployments across chains. i know gnosis safe does this as well. 👍